deploy.py

```diff
@@ -253,6 +256,24 @@ def handle_helm_upgrade():
     run_command(parameters)
 
 
+def get_local_ip() -> str:
+    """Determine the IP address of the current host using hostname -I."""
+    try:
+        result = subprocess.run(['hostname', '-I'],
+                              capture_output=True,
+                              text=True,
+                              check=True)
+
+        ip_addresses = result.stdout.strip().split()
+        if ip_addresses:
+            return ip_addresses[0]
+        return "127.0.0.1"
```

Comment:
Maybe actually we should not fallback here – we definitely shouldn't silently fallback which would be really confusing. I would just give an error.